app/controllers/application_controller.rb

```diff
@@ -477,6 +479,28 @@ def secure_session
     SecureSession.new(session["secure_session_id"] ||= SecureRandom.hex)
   end
 
+  def set_csp_header
+    nonce = SecureRandom.base64
+    request.env["nonce"] = nonce
+    response.headers["X-Discourse-CSP-Nonce"] = nonce
+
+    if Rails.env.development?
+      # don't do this in prod, vulnerable to Host/X-Forwarded-Host modification
+      base_url = request.protocol + request.host_with_port
```

Comment:
So that the CSP will work in local development, and produce a host of `localhost:3000` rather than just `localhost` (which will cause scripts to fail loading)